File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-202808-59615/workdir/src/main/java/surveycrawler/Boulder.java
package surveycrawler;

public class Boulder extends GroundType {

    // moving

    @Override
    public void moveCrawlerInDirection(SurveyCrawler aCrawler, Point aDirection) {
        aCrawler.signalBoulderFound();
    }

    // facing

    @Override
    public void turnCrawler(SurveyCrawler aCrawler, Runnable aTurn) {
        aCrawler.signalBoulderFound();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-212215-18237/analysis/own-tests-project/src/surveycrawler/boulder.py
from surveycrawler.ground_type import GroundType


class Boulder(GroundType):

    # moving

    def move_crawler_to(self, a_crawler, a_position, a_direction):
        a_crawler.signal_boulder_found()

    # facing

    def turn_crawler_with(self, a_crawler, a_turn):
        a_crawler.signal_boulder_found()
